### ISSUE TITLE
Persist training run history (hyperparameters, loss curve, duration)

### ISSUE THREAD

**Issue by will-lamerton** [Nano-Collective/nanotune#60]
## Problem

The live loss chart during `nanotune train` is React state in `src/commands/train.tsx`. When the command exits, it's gone.

Nothing is persisted about a training run — not the hyperparameters used, not the final loss, not the duration, not which base model produced which adapter. So you can't:

- compare two hyperparameter settings
- recall what produced the adapter currently sitting in `.nanotune/adapters/`
- plot loss after the fact
- tell whether a benchmark result came from the run you think it did

For a tool whose whole job is iterating on fine-tunes, this is a significant hole.

## Proposal

Write a run record per training invocation, e.g. `.nanotune/runs/<timestamp>.json`:

```json
{
  "timestamp": "...",
  "baseModel": "...",
  "training": { "iterations": 150, "learningRate": 5e-5, "...": "..." },
  "exampleCount": 702,
  "durationMs": 812000,
  "finalTrainLoss": 0.83,
  "finalValLoss": 0.91,
  "lossHistory": [ ... ],
  "adapterPath": "..."
}
```

Then surface it: `nanotune status` showing the last run's settings, and something like `nanotune runs` to list history.

## Acceptance criteria

- [ ] Each `train` invocation writes a run record, including runs stopped early
- [ ] Record captures hyperparameters, example count, duration, and loss history
- [ ] `status` shows what settings produced the current adapter
- [ ] Records are gitignored (they're per-machine artifacts)

## Related

Pairs naturally with checkpoint management — a run record should reference the checkpoints it produced.
